controllers/storagecluster/storagecluster_controller.go

```diff
@@ -155,6 +155,35 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 
+	enqueueFromStorageProfile := handler.EnqueueRequestsFromMapFunc(
+		func(_ context.Context, obj client.Object) []reconcile.Request {
+			spobj, ok := obj.(*ocsv1.StorageProfile)
+			if !ok {
+				spobj.Status.Phase = ocsv1.StorageProfilePhaseFailed
+				return []reconcile.Request{}
+			}
+
+			// Get the StorageCluster object
+			scList := &ocsv1.StorageClusterList{}
+			err := r.Client.List(r.ctx, scList, &client.ListOptions{Namespace: obj.GetNamespace(), Limit: 1})
+			if err != nil {
+				r.Log.Error(err, "Unable to list StorageCluster objects")
+				return []reconcile.Request{}
+			}
+
+			sc := scList.Items[0]
```

Comment:
If it reaches this point then I think it's safe to say it is a non-empty list:
> `List` retrieves list of objects for a given namespace and list options. On a successful call, `Items` field in the list will be populated with the result returned from the server.

Reply:
@raaizik I am not sure I agree. It can be that somebody created the SP CR before creating the SC CR. It is a valid configuration as the later creation of the CS CR will guarantee the reconciliation of the  SP CR as well (as they are reconciled by the same controller). 

We can avoid the issue by returning an empty reconcile request list.

```suggestion
			if len(scList) == 0 {
				return []reconcile.Request{}
			}
						
			sc := scList.Items[0]
```